bnm/recon/algo/service/subparcelation.py

```diff
@@ -53,83 +56,183 @@ def make_subparc(self, v, f, annot, roi_names, ctab, trg_area=100.0):
                 continue
 
             # compute area of faces in roi
-            roi_area = numpy.sum(self.surfaceService.tri_area(v[f[rfi]]))
+            roi_area = numpy.sum(self.surface_service.tri_area(surface.vertices[surface.triangles[rfi]]))
 
             # choose k for desired roi area
             k = int(roi_area / trg_area) + 1
             assert k >= 1
 
             # cluster centered vertices
-            v_roi = v[mask]
+            v_roi = surface.vertices[mask]
             _, i_lab = scipy.cluster.vq.kmeans2(v_roi - v_roi.mean(axis=0), k)
 
             # update annot
-            new_annot[mask] = next_aval + i_lab
+            new_annotation.region_mapping[mask] = next_aval + i_lab
             next_aval += k
-            new_names += ['%s-%d' % (name.decode('ascii'), j) for j in range(k)]
+            new_annotation.region_names += ['%s-%d' % (name.decode('ascii'), j) for j in range(k)]
 
         # create random colored ctab
-        new_ctab = numpy.random.randint(255, size=(len(new_names), 5))
-        r, g, b, _, _ = new_ctab.T
-        new_ctab[:, 3] = 0
-        new_ctab[:, 4] = r + 256 * g + 256 * 256 * b  # fs magic values
+        new_annotation.regions_color_table = numpy.random.randint(255, size=(len(new_annotation.region_names), 5))
+        r, g, b, _, _ = new_annotation.regions_color_table.T
+        new_annotation.regions_color_table[:, 3] = 0
+        new_annotation.regions_color_table[:, 4] = self.annotation_service.rgb_to_fs_magic_number([r, g, b])
 
-        return new_annot, new_ctab, new_names
+        return new_annotation
 
-    def subparc_files(self, hemi, parc_name, out_parc_name, trg_area):
+    def subparc_files(self, surf_path, annot_path, out_annot_parc_name, trg_area):
         trg_area = float(trg_area)
-        v, f = self.surfaceService.read_surf(hemi, 'sphere')
-        lab, ctab, names = read_annot(hemi, parc_name)
-        new_lab, new_ctab, new_names = self.make_subparc(v, f, lab, names, ctab, trg_area=trg_area)
-        write_annot(hemi, out_parc_name, new_lab, new_ctab, new_names)
+        surface = self.surface_service.surface_io.read(surf_path, False)
```

Comment:
This might be converted into `surface = self.surface_io.read(surf_path, False)` by adding an attribute of surface_io directly in this service.